viewing_party/party.py

```diff
@@ -1,23 +1,197 @@
+- Return a list of dictionaries, that represents a list of movies
+"""
+    list_of_user_data = []
+    list_of_friends_data = []
+    list_of_friends_data_comp = []
+    result = []
+    
+    for i in user_data["watched"]:
+        list_of_user_data.append(i)
+
+    for i in user_data["friends"]:
+        list_of_friends_data.append(i["watched"])
+
+    list_of_friends_data_comp = helper_join_list_of_lists_and_removes_duplicates(list_of_friends_data)
+    
+    for i in list_of_user_data:
+        if i not in list_of_friends_data_comp:
+            result.append(i)
+    
+    return result
+
+def get_friends_unique_watched(user_data):
+    """- Consider the movies that the user has watched, and consider the movies that their friends have watched. Determine which movies at least one of the user's friends have watched, but the user has not watched.
+- Return a list of dictionaries, that represents a list of movies
+"""
+    list_of_user_data = []
+    list_of_friends_data = []
+    list_of_friends_data_comp = []
+    result = []
+    
+    for i in user_data["watched"]:
+        list_of_user_data.append(i)
+
+    for i in user_data["friends"]:
+        list_of_friends_data.append(i["watched"])
+
+    list_of_friends_data_comp = helper_join_list_of_lists_and_removes_duplicates(list_of_friends_data)
+    
+    for i in list_of_friends_data_comp:
+        if i not in list_of_user_data:
+            result.append(i)
+    
+    return result
+
+# -----------------------------------------   
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_available_recs(user_data):
+    """- Determine a list of recommended movies. A movie should be added to this list if and only if:
+- The user has not watched it
+- At least one of the user's friends has watched
+- The `"host"` of the movie is a service that is in the user's `"subscriptions"`
+- Return the list of recommended movies"""
+    recommended =[]
+    user_hasnt_watched = get_friends_unique_watched(user_data)
 
+    for i in user_hasnt_watched:
+        if i["host"] in user_data["subscriptions"]:
+            recommended.append(i)
+    
+    return recommended
+# -----------------------------------------
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
+def get_new_rec_by_genre(user_data):
+    """  - The user has not watched it
+- At least one of the user's friends has watched
+- The `"genre"` of the movie is the same as the user's most frequent genre
+- Return the list of recommended movies"""
+    recommended = []
+    user_hasnt_watched = get_friends_unique_watched(user_data)
+    genres_list = []
+    
+    if user_data["watched"]:
+        for i in user_data["watched"]:
+            genres_list.append(i["genre"])
+        
+        frequent_genre = max(set(genres_list), key=genres_list.count)

```

Comment:
Could have leveraged the `get_most_watched_genre` function from the earlier wave here to avoid repetition!